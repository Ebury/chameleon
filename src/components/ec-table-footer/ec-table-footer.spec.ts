import { type ComponentMountingOptions, mount } from '@vue/test-utils';

import EcTableFooter from './ec-table-footer.vue';
import type { TableFooterProps } from './types';

describe('EcTableFooter', () => {
  function mountEcTableFooter(props?: Partial<TableFooterProps>, mountOpts?: ComponentMountingOptions<TableFooterProps>) {
    return mount(EcTableFooter, {
      props,
      ...mountOpts,
    });
  }

  it('should render with the given colspan', () => {
    const wrapper = mountEcTableFooter({
      colspan: 6,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render without footer slot', () => {
    const wrapper = mountEcTableFooter();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render with footer slot', () => {
    const wrapper = mountEcTableFooter({}, {
      slots: {
        default: '<p>Random text</p>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
