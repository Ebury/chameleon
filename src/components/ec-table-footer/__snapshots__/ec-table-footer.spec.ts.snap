// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EcTableFooter > should render with footer slot 1`] = `
<tfoot
  class="ec-table-footer"
  data-test="ec-table-footer"
>
  <tr>
    <td
      class="ec-table-footer__cell"
    >
      <div
        class="ec-table-footer__contents"
      >
        
        <p>
          Random text
        </p>
        
      </div>
    </td>
  </tr>
</tfoot>
`;

exports[`EcTableFooter > should render with the given colspan 1`] = `
<tfoot
  class="ec-table-footer"
  data-test="ec-table-footer"
>
  <tr>
    <td
      class="ec-table-footer__cell"
      colspan="6"
    >
      <div
        class="ec-table-footer__contents"
      >
        
        
      </div>
    </td>
  </tr>
</tfoot>
`;

exports[`EcTableFooter > should render without footer slot 1`] = `
<tfoot
  class="ec-table-footer"
  data-test="ec-table-footer"
>
  <tr>
    <td
      class="ec-table-footer__cell"
    >
      <div
        class="ec-table-footer__contents"
      >
        
        
      </div>
    </td>
  </tr>
</tfoot>
`;
